### ISSUE TITLE
Share dialog password field hidden when enabled

### ISSUE THREAD

**Issue by nickvergessen** [owncloud/core#15812]
### Steps to reproduce

1. Open the dialog
2. Share with Link
3. Check password box
4. Set password
5. Uncheck the link option
6. Share with link again

### Expected

The password field is either checked or collapsed

### Actual

The box is checked, but the password field is collapsed, so in order to enter a password you need to disable the password option… wait woot? :trollface:

**Comment by jancborchardt** [comment]
@nickvergessen if you assign @rullzer then you also need to mention him. I don’t think being assigned lets people know by mail or in any other way.

**Comment by jancborchardt** [comment]
(Ah, maybe it does ;)

**Comment by nickvergessen** [comment]
@jancborchardt yeah and we also have multiple communication channels, but yeah hard to follow it if you are not directly involved ;)

**Comment by jancborchardt** [comment]
So is this fixed with that first pull request @rullzer?

**Comment by rullzer** [comment]
@jancborchardt yes should be ;)
